test/clojure_mauth_client/middleware_test.clj

```diff
@@ -21,13 +33,36 @@
   (testing "Request should get validated successfully"
            (with-redefs [validate!   (fn [& _] (constantly true))]
              (let [f                     (middleware/wrap-mauth-verification mock-handler)
-                   {:keys [status body]} (f mock-post-request)]
+                   {:keys [status body]} (f mock-post-request-v2)]
                (is (= 200 status))
                (is (= "\"{\"test\":{\"response\":12345}}\"" body)))))
 
-  (testing "Request should get invalidated and should return Unauthorized with 401 error code"
+  (testing "Request should get invalidated and should return Unauthorized with 401 error code with v2"
            (with-redefs [validate!   (fn [& _] false)]
              (let [f                     (middleware/wrap-mauth-verification mock-handler)
-                   {:keys [status body]} (f mock-post-request)]
+                   {:keys [status body]} (f mock-post-request-v2)]
                (is (= 401 status))
-               (is (= "Unauthorized." body))))))
+               (is (= "Unauthorized." body)))))
+
+  (testing "Request should get invalidated and should return Unauthorized with 401 error code with v1"
+           (with-redefs [validate!   (fn [& _] false)]
+             (let [f                     (middleware/wrap-mauth-verification mock-handler)
```

Comment:
Could you provide a more meaningful name for the `f`?

Reply:
Also, can you add an Unauthorized test to v2 headers?

Reply:
Renamed f to request-function and we already have test case for Unauthorized  v2